source/common/ssl/context_impl.h

```diff
@@ -145,8 +147,13 @@ class ClientContextImpl : public ContextImpl, public ClientContext {
   bssl::UniquePtr<SSL> newSsl() const override;
 
 private:
+  int newSessionKey(SSL_SESSION* session);
+
   const std::string server_name_indication_;
   const bool allow_renegotiation_;
+  const size_t max_session_keys_;
+  mutable absl::Mutex session_keys_mu_;
+  mutable std::deque<bssl::UniquePtr<SSL_SESSION>> session_keys_ GUARDED_BY(session_keys_mu_);
```

Comment:
Make this mutable doesn't looks correct, should we just make `newSsl` non const? SslSocket holds non-const shared pointer so it should be OK.

Reply:
Fixed, thanks!